### DISCUSSION TITLE
Missing User Fields in Response When Using client.search_all_tweets

### DISCUSSION THREAD

**Discussion by AndrewQiu9748** [tweepy/tweepy#1777] [Questions] [answered]
Hi, I am using client.search_all_tweets from Api v2, but the user fields are missing, here is the code

for tweet in tweepy.Paginator(client.search_all_tweets, query=query,
                              tweet_fields=['author_id','context_annotations', 'created_at', 'geo', 'public_metrics','referenced_tweets'], 
                              start_time=start_time,
                              user_fields=['created_at','location','name'],
                              end_time=end_time, max_results=10).flatten(limit=100):
    print(tweet.id)
    print(tweet.author_id)
    print(tweet.text)
    print(tweet.context_annotations)
    print(tweet.created_at)
    print(tweet.geo)
    print(tweet.public_metrics)
    print(tweet.referenced_tweets)
    print(tweet.data)
    print(tweet.name)
    print(tweet.location)

I can get all the data except the 'name' and 'location'. And I have checked the tweet.data but cannot find the 'user_field'.

Q1:Where is the user_field, and how can I get the user location and user name.

Q2:The tweet.referenced_tweets only shows the tweet ID, how can I get the full text from the ID in the Paginator?

Can anyone help me? Many thanks for your time!

**Comment by Harmon758** [comment, marked as answer]
[User fields](https://tweepy.readthedocs.io/en/v4.4.0/client.html#user-fields) are returned as part of `includes`.
You'll need to use the `referenced_tweets.id` [expansion](https://tweepy.readthedocs.io/en/v4.4.0/client.html#expansions) to retrieve [Tweet fields](https://tweepy.readthedocs.io/en/v4.4.0/client.html#tweet-fields) for referenced Tweets.
Also see the [relevant FAQ section in the documentation about accessing `includes` data while using `Paginator`](https://tweepy.readthedocs.io/en/v4.4.0/faq.html#how-do-i-access-includes-data-while-using-paginator).

For code block usage, see https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks.

**Reply by AndrewQiu9748** [reply]
Thank you for the reply. 

Sorry, I still do not know how to access includes. Do I still need to keep the ```flatten()```, and iterate after it? Or ```flatten()``` iterates automatically and deletes the ```includes``` data? And what shall I do without it?

**Reply by 2jjk** [reply]
Hi. Did you manage to figure out how? Facing the same issue.

**Reply by AndrewQiu9748** [reply]
I figured out how to do it without```flatten()```, but now I do not know how to retrieve other data with ```limit()``` instead of flatten. It always shows "Rate limit exceeded". Here is the sample for username:
```
for response in tweepy.Paginator(client.search_all_tweets, query=query,
                              expansions=['author_id'],tweet_fields=['author_id','context_annotations', 'created_at', 'geo', 'public_metrics','referenced_tweets'], start_time=start_time,
                              end_time=end_time, max_results=10, limit=100):
    username = [tweet for tweet in response.includes["users"]]
    print(username)
    print(username[1])
```

**Reply by Harmon758** [reply]
> Sorry, I still do not know how to access includes. Do I still need to keep the `flatten()`, and iterate after it? Or `flatten()` iterates automatically and deletes the `includes` data? And what shall I do without it?

Again, see the [relevant FAQ section in the documentation about accessing `includes` data while using `Paginator`](https://tweepy.readthedocs.io/en/v4.4.0/faq.html#how-do-i-access-includes-data-while-using-paginator):
> `Paginator.flatten()` flattens the `data` and iterates over each object.
> To access `includes`, you’ll need to iterate through each response instead.

> It always shows "Rate limit exceeded".

There is also an [FAQ section in the documentation that addresses this as well](https://tweepy.readthedocs.io/en/v4.4.0/faq.html#why-am-i-getting-rate-limited-so-quickly-when-using-client-search-all-tweets-with-paginator).
